PurchasesCoreSwift/Purchasing/StoreKitRequestFetcher.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  StoreKitRequestFetcher.swift
+//  PurchasesCoreSwift
+//
+//  Created by Andrés Boedo on 6/29/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+@objc(RCReceiptRefreshRequestFactory) public class ReceiptRefreshRequestFactory: NSObject {
+
+    @objc public func receiptRefreshRequest() -> SKReceiptRefreshRequest {
+        return SKReceiptRefreshRequest()
+    }
+}
+
+@objc(RCStoreKitRequestFetcher) public class StoreKitRequestFetcher: NSObject {
+    private let requestFactory: ReceiptRefreshRequestFactory
+    private var receiptRefreshRequest: SKRequest?
+    private var receiptRefreshCompletionHandlers: [() -> Void]
+    private let queue = DispatchQueue(label: "StoreKitRequestFetcher")
+
+    @objc public init(requestFactory: ReceiptRefreshRequestFactory = ReceiptRefreshRequestFactory()) {
+        self.requestFactory = requestFactory
+        receiptRefreshRequest = nil
+        receiptRefreshCompletionHandlers = []
+    }
+
+    @objc public func fetchReceiptData(_ completion: @escaping () -> Void) {
+        queue.async {
```

Comment:
this bit is technically a change in behavior - we were using `@synchronized` for this, but I don't think there's a clear equivalent in Swift. We could do `queue.sync`, but that might introduce deadlocks if this method gets called from the main thread.